File: Sc.Models/Dtos/ApproveRegistrations/Base/BaseApproveRegistrationDto.cs
using Sc.Models.Dtos.Auth;
using Sc.Models.Dtos.Suppliers;
using Sc.Models.Entities.Base;
using Sc.Models.Enums.State;

namespace Sc.Models.Dtos.ApproveRegistrations.Base
{
    public abstract class BaseApproveRegistrationDto : Entity
    {
        public DateTime CreateDate { get; set; }

        public DateTime FinishDate { get; set; }

        public int AdministratedUserId { get; set; }

        public string AdministratedUsername { get; set; }

        public ApproveRegistrationState State { get; set; }

        public string DeclinedNote { get; set; }

        public SignUpDto SignUpDto { get; set; }

        public int? SupplierId { get; set; }
        public SupplierDto Supplier { get; set; }
    }
}

File: WebApplication/src/approve-registrations/dtos/base/base-approve-registration.dto.ts
import { ApproveRegistrationState } from "src/approve-registrations/enums/approve-registration-state.enum";
import { SignUpDto } from "src/auth/dtos/sign-up/sign-up.dto";

export class BaseApproveRegistrationDto {
    id: number;
    createDate: Date;
    finishDate: Date;
    administratedUserId: number;
    administratedUsername: string;
    state: ApproveRegistrationState;
    declinedNote: string;
    signUpDto: SignUpDto;
    supplierId: number;
}